internal/controller/ratelimit_workflow_helpers.go

```diff
@@ -54,7 +55,23 @@ func GetLimitadorFromTopology(topology *machinery.Topology) *limitadorv1alpha1.L
 	}
 
 	limitador := limitadorObj.(*controller.RuntimeObject).Object.(*limitadorv1alpha1.Limitador)
-	return limitador
+	return limitador.DeepCopy()
+}
+
+func GetMachineryObjectFromTopology(topology *machinery.Topology, groupkind schema.GroupKind) machinery.Object {
```

Comment:
This seems possibly in the wrong place? Seems a quite useful utility that might be used in many locations?

Reply:
moved